mdns.go

```diff
@@ -31,21 +54,76 @@ func publishHomeAssistant() {
 	}
 }
 
-// Get preferred outbound ip of this machine
-// https://stackoverflow.com/a/37382208
-// Can be removed after Supervisor v248 with new lookup
-func getOutboundIP() net.IP {
-	conn, err := net.Dial("udp", "8.8.8.8:80")
+// Get the first IPv4 address from the interface information
+func getFirstIPv4Address(iface map[string]any) (net.IP, error) {
+	ipv4Data, ok := iface["ipv4"].(map[string]any)
+	if !ok {
+		return nil, fmt.Errorf("no ipv4 addresses found")
+	}
+
+	addresses, ok := ipv4Data["address"].([]any)
+	if !ok || len(addresses) == 0 {
+		return nil, fmt.Errorf("empty ipv4 address list")
+	}
+
+	addr, ok := addresses[0].(string)
+	if !ok {
+		return nil, fmt.Errorf("can't parse first ipv4 address")
+	}
+
+	ip, _, err := net.ParseCIDR(addr)
+
 	if err != nil {
-		log.Fatal(err)
+		return nil, fmt.Errorf("invalid IPv4 address: %s", err)
 	}
-	defer func() {
-		if err := conn.Close(); err != nil {
-			log.Printf("error closing mdns connection: %v", err)
+
+	return ip, nil
+}
+
+// Get IP address of the default interface from Supervisor
+func getOutboundIP() (net.IP, error) {
+	supervisorHost := getSupervisorHost()
+
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", "http://"+supervisorHost+"/network/interface/default/info", nil)
+
+	if err != nil {
+		return nil, fmt.Errorf("can't create request to Supervisor: %s", err)
+	}
+
+	req.Header.Add("Authorization", "Bearer "+os.Getenv("SUPERVISOR_TOKEN"))
```

Comment:
Lets make this a helper function similar to `getSupervisorHost()`

Reply:
You mean something like `setSupervisorAuthHeader(req)`?